readthedocs/embed/v3/views.py

```diff
@@ -138,7 +140,11 @@ def _parse_based_on_doctool(self, page_content, fragment, doctool, doctoolversio
 
         node = None
         if fragment:
-            selector = f'#{fragment}'
+            # NOTE: we use the `[id=]` selector because using `#{id}` requires
+            # scaping the selector since CSS does not support the same
+            # characters than the `id=` HTML attribute
+            # https://www.w3.org/TR/CSS21/syndata.html#value-def-identifier
```

Comment:
💯  comment.

Reply:
@nienn helped me here to understand why this happened 💪🏼